plugins/inputs/supervisor/supervisor.go

```diff
@@ -0,0 +1,170 @@
+package supervisor
+
+import (
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"github.com/kolo/xmlrpc"
+	"net/url"
+)
+
+type Supervisor struct {
+	Log telegraf.Logger `toml:"-"`
+
+	Server         string `toml:"url"`
+	PidGather      bool   `toml:"gather_pid"`
+	ExitCodeGather bool   `toml:"gather_exit_code"`
+	UseIdentTag    bool   `toml:"use_identification_tag"`
+
+	Status SupervisorInfo
+}
+
+type ProcessInfo struct {
+	Name          string `xmlrpc:"name"`
+	Group         string `xmlrpc:"group"`
+	Description   string `xmlrpc:"description"`
+	Start         int32  `xmlrpc:"start"`
+	Stop          int32  `xmlrpc:"stop"`
+	Now           int32  `xmlrpc:"now"`
+	State         int16  `xmlrpc:"state"`
+	Statename     string `xmlrpc:"statename"`
+	StdoutLogfile string `xmlrpc:"stdout_logfile"`
+	StderrLogfile string `xmlrpc:"stderr_logfile"`
+	SpawnErr      string `xmlrpc:"spawnerr"`
+	ExitStatus    int8   `xmlrpc:"exitstatus"`
+	Pid           int32  `xmlrpc:"pid"`
+}
+
+type SupervisorInfo struct {
+	StateCode int8   `xmlrpc:"statecode"`
+	StateName string `xmlrpc:"statename"`
+	Ident     string
+}
+
+const sampleConfig = `
+  ## Url of supervisor's XML-RPC endpoint if basic auth enabled in supervisor http server,
+  ## than you have to add credentials to url (ex. http://login:pass@localhost:9001/RPC2)
+  # url="http://localhost:9001/RPC2"
+  ## Use supervisor identification string as server tag
+  use_identification_tag = false
+  ## Gather PID of running processes
+  gather_pid = false
+  ## Gather exit codes of processes
+  gather_exit_code = false
+`
+
+func (s *Supervisor) Description() string {
+	return "Gather info about processes state, that running under supervisor using its XML-RPC API"
+}
+
+func (s *Supervisor) SampleConfig() string {
+	return sampleConfig
+}
+
+func (s *Supervisor) Gather(acc telegraf.Accumulator) error {
+	//Initializing XML-RPC client
+	client, err := xmlrpc.NewClient(s.Server, nil)
+	if err != nil {
+		return err
+	}
+	defer client.Close()
+
+	//API call to get information about all running processes
+	var rawProcessData []ProcessInfo
+	err = client.Call("supervisor.getAllProcessInfo", nil, &rawProcessData)
+	if err != nil {
+		return err
+	}
+
+	//API call to get information about instance status
+	err = client.Call("supervisor.getState", nil, &s.Status)
+	if err != nil {
+		return err
+	}
+
+	//API call to get identification string
+	err = client.Call("supervisor.getIdentification", nil, &s.Status.Ident)
+	if err != nil {
+		return err
+	}
+
+	//Iterating through array of structs with processes info and adding fields to accumulator
+	for _, process := range rawProcessData {
+		processTags, processFields, err := s.parseProcessData(process)
+		if err != nil {
+			return err
+		}
+		acc.AddFields("supervisor_processes", processFields, processTags)
+	}
```

Comment:
Does it make sense to process the remaining data even if one of them fails? If so you should do
```suggestion
	for _, process := range rawProcessData {
		processTags, processFields, err := s.parseProcessData(process)
		if err != nil {
			acc.AddError(err)
			continue
		}
		acc.AddFields("supervisor_processes", processFields, processTags)
	}
```

Reply:
Can you please comment on this?

Reply:
Your approach fits better, i will include this in next updates.